tests/epyccel/recognised_functions/test_math_funcs.py

```diff
@@ -377,12 +382,735 @@ def floor_return_type_real(x):
+    '''test type copysign(real, real) => should return real number'''
+    @types('real', 'real')
+    def copysign_return_type(x, y):
+        from math import copysign
+        a = copysign(x, y)
+        return a
+
+    f1 = epyccel(copysign_return_type, language = language)
+    x = rand() # real
+    y = rand() # real
+
+    # Same sign
+    assert(type(f1(x, y)) == type(copysign_return_type(x, y)))
+    assert(type(f1(-x, -y)) == type(copysign_return_type(-x, -y)))
+    # Different sign
+    assert(type(f1(-x, y)) == type(copysign_return_type(-x, y)))
+    assert(type(f1(x, -y)) == type(copysign_return_type(x, -y)))
+
+def test_copysign_return_type_2(language = 'c'): # copysign
+    '''test type copysign(int, int) => should return real type'''
+    @types('int', 'int')
+    def copysign_return_type(x, y):
+        from math import copysign
+        a = copysign(x, y)
+        return a
+
+    f1 = epyccel(copysign_return_type, language = language)
+    high = 10000000
+    x = randint(high)   # int
+    y = randint(high)   # int
+
+    # Same sign
+    assert(type(f1(x, y)) == type(copysign_return_type(x, y)))
+    assert(type(f1(-x, -y)) == type(copysign_return_type(-x, -y)))
+    # Different sign
+    assert(type(f1(-x, y)) == type(copysign_return_type(-x, y)))
+    assert(type(f1(x, -y)) == type(copysign_return_type(x, -y)))
+
+def test_copysign_return_type_3(language = 'c'): # copysign
+    '''test type copysign(int, real) => should return real type'''
+    @types('int', 'real')
+    def copysign_return_type(x, y):
+        from math import copysign
+        a = copysign(x, y)
+        return a
+
+    f1 = epyccel(copysign_return_type, language = language)
+    high = 10000000
+    x = randint(high)   # int
+    y = rand()          # real
+
+    # Same sign
+    assert(type(f1(x, y)) == type(copysign_return_type(x, y)))
+    assert(type(f1(-x, -y)) == type(copysign_return_type(-x, -y)))
+    # Different sign
+    assert(type(f1(-x, y)) == type(copysign_return_type(-x, y)))
+    assert(type(f1(x, -y)) == type(copysign_return_type(x, -y)))
+
+def test_copysign_return_type_4(language = 'c'): # copysign
+    '''test type copysign(real, int) => should return real type'''
+    @types('real', 'int')
+    def copysign_return_type(x, y):
+        from math import copysign
+        a = copysign(x, y)
+        return a
+
+    f1 = epyccel(copysign_return_type, language = language)
+    high = 10000000
+    x = rand()          # real
+    y = randint(high)   # int
+
+    # Same sign
+    assert(type(f1(x, y)) == type(copysign_return_type(x, y)))
+    assert(type(f1(-x, -y)) == type(copysign_return_type(-x, -y)))
+    # Different sign
+    assert(type(f1(-x, y)) == type(copysign_return_type(-x, y)))
+    assert(type(f1(x, -y)) == type(copysign_return_type(x, -y)))
+
+
+#----------------------------- isfinite function -----------------------------#
+
+@pytest.mark.xfail(reason = "pyccel non implemented data type issues #388-390")
```

Comment:
Issues #388, #389, #390 have now been fixed. This xfail mark should therefore be removed